### PR TITLE
docs: migrate README to Sphinx

### PR DESCRIPTION
**Pull request by jia200x**
This PR moves parts of the README.md to Sphinx.

This will help us to provide a better user experience when the documentation is hosted on "Read The Docs".

I also updated the `index.rst` file to add new sections and moved the API to its own section.

The idea is to use this as a template for writing more documentation.

### CODE REVIEW

**Comment by felipehuerta17** [comment]
I just built this and it works very well

File: .gitignore [hunk 1]
```diff
@@ -19,6 +19,7 @@ __pycache__/
 .Python
 build/
 _build/
+doc/build/
```
Comment (sasalatart) [inline .gitignore:22]: Not 100% sure, but I thought that this was covered by https://github.com/open-sir/open-sir/pull/42/files#diff-a084b794bc0759e7a6b77810e01874f2R20

File: Pipfile [hunk 2]
```diff
@@ -28,4 +28,4 @@ allow_prereleases = true
 start = "python opensir-cli"
 lint = "pylint **/*.py"
 test = "pytest"
-doc = "sphinx-build -M html . _build -W"
```
Comment (sasalatart) [inline Pipfile:31]: Does this mean we can remove the `_build` entry from the gitignore?

File: doc/getting-started.rst [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+::
+
+    pip uninstall opensir
+
+Usage example
+*************
+
+You can use Open-SIR to create a 6 days prediction of the number of susceptible (S), infected (I) and removed (R) population. 
+The initial conditions `-i` represent Ealing data as of 04/04/2020. The parameters `-p` provide a prediction in the hypothetical 
+case that no lockdown would be taking place.
```
Comment (sasalatart) [inline doc/getting-started.rst:32]: Just curious, how much chars per line are you using?

File: doc/getting-started.rst [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                            CSV delimiter
+
+The input file is a TOML file with the following format
+::
+
+    [initial_conds]
+    <cond> = value
+
+    [parameters]
+    <paran> = value
```
Comment (sasalatart) [inline doc/getting-started.rst:61]: Should this be `<param>` instead?

File: Pipfile (hunk 2 above)
Comment (jia200x) [reply Pipfile:31]: Yes, you are right. We can remove it

File: doc/getting-started.rst (hunk 3 above)
Comment (jia200x) [reply doc/getting-started.rst:32]: it should be 80, but I forgot to apply formats here. I can adapt it :)

File: doc/getting-started.rst (hunk 4 above)
Comment (jia200x) [reply doc/getting-started.rst:61]: nice catch! thanks

**Comment by jia200x** [comment]
@sasalatart @felipehuerta17 I just pushed all fixes and improved a little bit the getting started section.

Please let me know if there's something I forgot. If not, I can proceed to squash and rebase

**Comment by jia200x** [comment]
> Thinking on the trees, below contents we may have:
    Model class
    -- Model functions
    --- solve
    --- fit()
    -- Models
    --- SIR model
    --- SIR-X model
    Post regression analysis
    -- ci_bootstrap
    -- ci_block_cv
    Tutorials
And we can link to the Jupyter Notebook tutorials from the documentation :)

I like this approach.
We found a promising Sphinx library to use Jupyter notebooks: https://nbsphinx.readthedocs.io/en/0.6.0/
I guess that's a safe bet

**Pull request merged**